conans/client/recipe_hash.py

```diff
@@ -0,0 +1,32 @@
+import os
+from conans.util.files import load
+from conans.util.sha import sha1
+
+
+def _normalize_crlfs(path):
+    contents = load(path, binary=True)
+    return contents.replace(b'\r\n', b'\n').replace(b'\r', b'\n')
+
+
+def _get_file_digest(path):
+    return sha1(_normalize_crlfs(path))
+
+
+def get_normalized_hash(folder):
+    paths = []
+    for root, _, files in os.walk(folder):
+        for name in files:
+            if not name.endswith(".pyc"):
```

Comment:
From Manifest:
```python
file_dict.pop(".DS_Store", None)  # Exclude tmp in mac
```
seems that such file have to be ignored from the hash. In fact the full implementation is:

```python
from conans.paths import CONAN_MANIFEST, CONANFILE
        file_dict.pop(PACKAGE_TGZ_NAME, None)  # Exclude the PACKAGE_TGZ_NAME
        file_dict.pop(EXPORT_TGZ_NAME, None)  # Exclude the EXPORT_TGZ_NAME
        file_dict.pop(CONAN_MANIFEST, None)  # Exclude the MANIFEST itself
        file_dict.pop(CONANFILE + "c", None)  # Exclude the CONANFILE.pyc
        file_dict.pop(".DS_Store", None)  # Exclude tmp in mac

        file_dict = {key: value for key, value in file_dict.items()
                     if not key.startswith("__pycache__")}
```